llama_index/embeddings/gradient.py

```diff
@@ -0,0 +1,98 @@
+        `pip install gradientai`.
+    """
+
+    _gradient: Any = PrivateAttr()
+    _model: Any = PrivateAttr()
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "GradientEmbedding"
+
+    def __init__(
+        self,
+        *,
+        gradient_model_slug: str,
+        gradient_access_token: Optional[str] = None,
+        gradient_workspace_id: Optional[str] = None,
+        gradient_host: Optional[str] = None,
+        **kwargs,
+    ):
+        """Initializes the GradientEmbedding class.
+
+        During the initialization the `gradientai` package is imported. Using the access token,
+        workspace id and the slug of the model, the model is fetched from Gradient AI and prepared to use.
+
+        Args:
+            gradient_model_slug (str): The model slug of the model in the Gradient AI account.
+            gradient_access_token (str, optional): The access token of the Gradient AI account, if
+                `None` read from the environment variable `GRADIENT_ACCESS_TOKEN`.
+            gradient_workspace_id (str, optional): The workspace ID of the Gradient AI account, if `None`
+                read from the environment variable `GRADIENT_WORKSPACE_ID`.
+            gradient_host (str, optional): The host of the Gradient AI API. Defaults to None, which
+              means the default host is used.
+
+        Raises:
+            ImportError: If the `gradientai` package is not available in the PYTHONPATH.
+            ValueError: If the model cannot be fetched from Gradient AI.
+        """
+        try:
+            import gradientai
+        except ImportError:
+            raise ImportError("GradientEmbedding requires `pip install gradientai`.")
+
+        self._gradient = gradientai.Gradient(
+            access_token=gradient_access_token,
+            workspace_id=gradient_workspace_id,
+            host=gradient_host,
+        )
+
+        try:
+            self._model = self._gradient.get_embeddings_model(slug=gradient_model_slug)
+        except gradientai.openapi.client.exceptions.UnauthorizedException as e:
+            logger.error(f"Error while loading model {gradient_model_slug}.")
+            self._gradient.close()
+            raise ValueError("Unable to fetch the requested embeddings model") from e
+
+        super().__init__(model_name=gradient_model_slug, **kwargs)
+
+    async def _aget_query_embedding(self, query: str) -> Embedding:
+        # Gradient AI doesn't have the proper API for async yet, so we just use the sync version.
+        return self._get_query_embedding(query)
+
+    def _get_text_embedding(self, text: str) -> Embedding:
+        return self._get_text_embeddings([text])[0]
+
+    def _get_text_embeddings(self, texts: List[str]) -> List[Embedding]:
+        inputs = [{"input": text} for text in texts]
+
+        result = self._model.generate_embeddings(inputs=inputs).embeddings
```

Comment:
question to the team: currently gradient supports up to 100 inputs, does it make sense to move around some batching logic here?

Reply:
Actually, I think you can re-declare the `embed_batch_size` attribute and specify constraints

```
from llama_index.bridge.pydantic import Field
from llama_index.embeddings.base import DEFAULT_EMBED_BATCH_SIZE

class GradientEmbedding(BaseEmbedding):
    embed_batch_size: int = Field(default=DEFAULT_EMBED_BATCH_SIZE, gt=0, lte=100)
    ....

Reply:
looks like it worked, thanks!